### PR TITLE
Gf 301 reviews and news refactor components

### PR DESCRIPTION
**Pull request by Additivemenu**
refactor done, now we could use a reusable component to generate both 'AllReviews' component in review page and 'LatestNews' in news page.  
Screenshot 2023-03-03 at 10 02 58 pm
Screenshot 2023-03-03 at 10 03 19 pm

### CODE REVIEW

File: src/components/NewsPageV2/LatestNews/LatestNews.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    margin: 30px auto;
+    width: 130px;
+    font-weight: bold;
+    &:hover {
+      background-color: #70e702;
+    }
+  }
+`;
+
+interface LatestNews {
```
Comment (weited) [inline src/components/NewsPageV2/LatestNews/LatestNews.tsx]: interface 放到interfaces文件夹里面去  取名大写字母 I 开头 , 比如 interface ILatestNews  然后复用

File: src/components/NewsPageV2/LatestNews/LatestNews.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  subtitle: string;
+  author: string;
+  likeCount: number;
+  commentCount: number;
+}
+
+const LatestNews = () => {
+  const [latestNewsList, setLatestNewsList] = useState<LatestNews[]>([]);
+
+  const hasMountedRef = useRef(false);
```
Comment (weited) [inline src/components/NewsPageV2/LatestNews/LatestNews.tsx]: 这个用来做什么啊 下面没看到用?

File: src/components/NewsPageV2/LatestNews/LatestNewsItem/LatestNewsItem.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    likeCount: number;
+    commentCount: number;
+    game?: {
+      id: string;
+      name: string;
+    };
+  };
+}
+
+const LatestNewsItem = ({ news }: Props) => {
```
Comment (weited) [inline src/components/NewsPageV2/LatestNews/LatestNewsItem/LatestNewsItem.tsx]: 需要destruct 吗

File: src/components/NewsPageV2/LatestNews/PlatformSelectionBar/PlatformSelectionBar.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+const OuterContainer = styled.div`
+  display: flex;
+  margin-left: 135px;
+  margin-top: 25px;
+`;
+
+const CustomButton = styled(Button)`
+  && {
+    background-color: #222430;
```
Comment (weited) [inline src/components/NewsPageV2/LatestNews/PlatformSelectionBar/PlatformSelectionBar.tsx]: 调用主题变量?

File: src/components/NewsPageV2/NewsPageV2.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+//   daysAndOverview: string;
+//   game: string;
+//   author: string;
+//   commNum: string;
+//   likeNum: string;
+// };
+// const initialState: ReviewCardType[] = [];
+// ! --------------------------------------------------------------------------------------------------
+
+type ArticleFormat = {
```
Comment (weited) [inline src/components/NewsPageV2/NewsPageV2.tsx]: 上面用interface的话 这里也统一用interface   直接调用上面那个

File: src/components/NewsPageV2/NewsPageV2.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  mockNewsItem,
+  mockNewsItemWithGame,
+  mockNewsItem,
+  mockNewsItem,
+];
+
+// NewsPageV2 component ======================================================================
+const NewsPageV2 = () => {
+  // hooks and states -------------------
+  const articleType = 'News'; // which is used to differentiate review page and news page
```
Comment (weited) [inline src/components/NewsPageV2/NewsPageV2.tsx]: 写成 enum

File: src/components/NewsPageV2/components/LatestNews/components/LatestNewsCard/LatestNewsCard.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
+import { Box, Typography } from '@mui/material';
+import React from 'react';
+import PersonRoundedIcon from '@mui/icons-material/PersonRounded';
+import FavoriteRoundedIcon from '@mui/icons-material/FavoriteRounded';
+import ChatRoundedIcon from '@mui/icons-material/ChatRounded';
+import style from './index.module.css';
+import theme from '@/styles/theme';
+
+type Props = {
```
Comment (weited) [inline src/components/NewsPageV2/components/LatestNews/components/LatestNewsCard/LatestNewsCard.tsx]: 统一一下用interface?

File: src/components/Reusable_Components/ArticlesShowCase/ArticlesShowCase.tsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  & h2 {
+    font-family: Gen Jyuu Gothic Monospace;
+    font-size: 28px;
+    font-weight: 800;
+    text-transform: uppercase;
+    letter-spacing: 0em;
+  }
+`;
+
+type ArticleFormat = {
```
Comment (weited) [inline src/components/Reusable_Components/ArticlesShowCase/ArticlesShowCase.tsx]: 复用interface

File: src/components/Reusable_Components/ArticlesShowCase/components/SelectionBars/components/PlatformBars/PlatformBars.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+`;
+
+interface PlatformBarsProps {
+  platformSelected: string;
+  setPlatformSelected: React.Dispatch<React.SetStateAction<string>>;
+}
+
+const PlatformBars = ({ platformSelected, setPlatformSelected }: PlatformBarsProps) => {
+  // hooks and states
+  const Platforms = ['All', 'PlayStation', 'Nintendo', 'Xbox', 'PC'];
```
Comment (weited) [inline src/components/Reusable_Components/ArticlesShowCase/components/SelectionBars/components/PlatformBars/PlatformBars.tsx]: 放在constant文件夹里  复用constant

File: src/components/Reusable_Components/ArticlesShowCase/components/ShowCaseBody/ShowCaseBody.tsx [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+    setSelectedPlatformArticleList([...selectedPlatformArticleList, ...addedArticles]);
+  };
+  // !--------------------------------------------------------------------------------------------------------
+
+  // jsx ---------------------------------------
+
+  return (
+    <Container>
+      {selectedPlatformArticleList.map((article: ArticleFormat) => {
+        const { coverUrl, title, date, subtitle, author, commentCount, likeCount, game } = article;
```
Comment (weited) [inline src/components/Reusable_Components/ArticlesShowCase/components/ShowCaseBody/ShowCaseBody.tsx]: 要不要直接把article传到卡里 ?  不用在这里解构

**Review by weited — COMMENTED** [review]
注意统一一下interface提出来复用

File: src/components/Reusable_Components/ArticlesShowCase/components/ShowCaseBody/ShowCaseBody.tsx (hunk 10 above)
Comment (Additivemenu) [reply src/components/Reusable_Components/ArticlesShowCase/components/ShowCaseBody/ShowCaseBody.tsx]: 这里我试了下article传到map()的arguemnt里解构, eslint会莫名其妙地报错:
ERROR in [eslint] 
src/components/Reusable_Components/ArticlesShowCase/components/ShowCaseBody/ShowCaseBody.tsx
  Line 44:3:  'coverUrl' PropType is defined but prop is never used      react/no-unused-prop-types
  Line 45:3:  'title' PropType is defined but prop is never used         react/no-unused-prop-types
  Line 46:3:  'date' PropType is defined but prop is never used          react/no-unused-prop-types
  Line 47:3:  'subtitle' PropType is defined but prop is never used      react/no-unused-prop-types
  Line 48:3:  'author' PropType is defined but prop is never used        react/no-unused-prop-types
  Line 49:3:  'likeCount' PropType is defined but prop is never used     react/no-unused-prop-types
  Line 50:3:  'commentCount' PropType is defined but prop is never used  react/no-unused-prop-types
  Line 51:3:  'game' PropType is defined but prop is never used          react/no-unused-prop-types

Search for the keywords to learn more about each error.

File: src/components/NewsPageV2/LatestNews/LatestNews.tsx (hunk 1 above)
Comment (Additivemenu) [reply src/components/NewsPageV2/LatestNews/LatestNews.tsx]: LatestNews用复用的组件写, 这是之前遗留的, 已经删除

File: src/components/NewsPageV2/LatestNews/LatestNews.tsx (hunk 2 above)
Comment (Additivemenu) [reply src/components/NewsPageV2/LatestNews/LatestNews.tsx]: LatestNews用复用的组件写, 这是之前遗留的, NewsPageV2中已经删除

File: src/components/NewsPageV2/LatestNews/LatestNewsItem/LatestNewsItem.tsx (hunk 3 above)
Comment (Additivemenu) [reply src/components/NewsPageV2/LatestNews/LatestNewsItem/LatestNewsItem.tsx]: LatestNews用复用的组件写, 这是之前遗留的, NewsPageV2中已经删除

File: src/components/NewsPageV2/LatestNews/PlatformSelectionBar/PlatformSelectionBar.tsx (hunk 4 above)
Comment (Additivemenu) [reply src/components/NewsPageV2/LatestNews/PlatformSelectionBar/PlatformSelectionBar.tsx]: LatestNews用复用的组件写, 这是之前遗留的, NewsPageV2中已经删除

File: src/components/NewsPageV2/components/LatestNews/components/LatestNewsCard/LatestNewsCard.tsx (hunk 7 above)
Comment (Additivemenu) [reply src/components/NewsPageV2/components/LatestNews/components/LatestNewsCard/LatestNewsCard.tsx]: LatestNews用复用的组件写, 这是之前遗留的, 现在NewsPageV2中已经删除

**Pull request merged**